src/test/java/com/stripe/functional/BalanceTransactionTest.java

```diff
@@ -1,34 +1,43 @@
 package com.stripe.functional;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 
-import com.stripe.BaseStripeFunctionalTest;
+import com.stripe.BaseStripeTest;
 import com.stripe.exception.StripeException;
 import com.stripe.model.BalanceTransaction;
 import com.stripe.model.BalanceTransactionCollection;
-import com.stripe.model.Charge;
+import com.stripe.net.APIResource;
 
 import java.util.HashMap;
+import java.util.Map;
 
 import org.junit.Test;
 
-public class BalanceTransactionTest extends BaseStripeFunctionalTest {
+public class BalanceTransactionTest extends BaseStripeTest {
+  public static final String RESOURCE_ID = "bt_123";
+
   @Test
-  public void testBalanceTransactionRetrieval() throws StripeException {
-    Charge.create(defaultChargeParams);
-    BalanceTransactionCollection balanceTransactions = BalanceTransaction.all(null);
-    assertFalse(balanceTransactions.getData().isEmpty());
-    BalanceTransaction first = balanceTransactions.getData().get(0);
-    assertNotNull(first.getStatus());
-
-    HashMap<String, Object> fetchParams = new HashMap<String, Object>();
-    fetchParams.put("count", 2);
-    assertEquals(BalanceTransaction.all(fetchParams).getData().size(), 2);
-
-    BalanceTransaction retrieved = BalanceTransaction.retrieve(first.getId());
-    assertEquals(retrieved.getId(), first.getId());
-    assertEquals(retrieved.getSource(), first.getSource());
+  public void testRetrieve() throws StripeException {
+    BalanceTransaction resource = BalanceTransaction.retrieve(RESOURCE_ID);
+
+    assertNotNull(resource);
+    verifyRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/balance/history/%s", RESOURCE_ID)
+    );
+  }
+
+  @Test
+  public void testList() throws StripeException {
+    Map<String, Object> listParams = new HashMap<String, Object>();
+    listParams.put("limit", 1);
+
+    BalanceTransactionCollection resources = BalanceTransaction.list(listParams);
+
+    assertNotNull(resources);
+    verifyRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/balance/history")
```

Comment:
should we verify that params are sent? (won't repeat it for all retrieve, might be worth fixing all)

Reply:
same as above, we are explicitly passing `limit` but not verifying it.